x-pack/plugins/security_solution/server/endpoint/routes/trusted_apps/trusted_apps.test.ts

```diff
@@ -24,15 +24,23 @@ import {
 import { xpackMocks } from '../../../../../../mocks';
 import { ENDPOINT_TRUSTED_APPS_LIST_ID } from '../../../../../lists/common/constants';
 import { EndpointAppContext } from '../../types';
-import { ExceptionListClient } from '../../../../../lists/server';
-import { getExceptionListItemSchemaMock } from '../../../../../lists/common/schemas/response/exception_list_item_schema.mock';
+import { ExceptionListClient, ListClient } from '../../../../../lists/server';
+import { listMock } from '../../../../../lists/server/mocks';
 import { ExceptionListItemSchema } from '../../../../../lists/common/schemas/response';
 import { DeleteTrustedAppsRequestParams } from './types';
+import { getExceptionListItemSchemaMock } from '../../../../../lists/common/schemas/response/exception_list_item_schema.mock';
+
+type RequestHandlerContextWithLists = ReturnType<typeof xpackMocks.createRequestHandlerContext> & {
+  lists?: {
```

Comment:
@madirey , @FrankHassanabad ,
I would like to define this `lists` type here by referencing a Type in Lists plugin. You currently have this type defined in `server/types.ts`:

https://github.com/elastic/kibana/blob/fe1c508d8d42f750f830fc2e3caec5aaa4f66c0d/x-pack/plugins/lists/server/types.ts#L43-L46

Would you be ok if I break out the object that this `Promise<>` wraps into a separate `interface` so that I can reference it? so it would be something like:

```typescript
export interface ListsRequestHandlerContext {
  getListClient: () => ListClient;
  getExceptionListClient: () => ExceptionListClient;
}

export type ContextProviderReturn = Promise<ListsRequestHandlerContext>;
```

Reply:
That's fine with me, but I'd get @FrankHassanabad or @yctercero to sign off...

Reply:
Thanks for the feedback @madirey :)

If @FrankHassanabad does get some time to comment here 😬 , I have also been wondering the following: 
I see that the `context` provided by Kibana server to each route handler some how "knows" about the `lists` namespace in it although the `RequestHandler<>` generic type does not have a way to set it. Is it because of this definition in the `server/types`:

https://github.com/elastic/kibana/blob/fe1c508d8d42f750f830fc2e3caec5aaa4f66c0d/x-pack/plugins/lists/server/types.ts#L47-L54